bfvmm/src/vmcs/src/vmcs_intel_x64_debug.cpp

```diff
@@ -22,524 +23,716 @@
+    dump_vmcs_field(guest_ldtr_limit::addr, guest_ldtr_limit::name, guest_ldtr_limit::exists());
+    dump_vmcs_field(guest_tr_limit::addr, guest_tr_limit::name, guest_tr_limit::exists());
+    dump_vmcs_field(guest_gdtr_limit::addr, guest_gdtr_limit::name, guest_gdtr_limit::exists());
+    dump_vmcs_field(guest_idtr_limit::addr, guest_idtr_limit::name, guest_idtr_limit::exists());
+    dump_vmcs_field(guest_es_access_rights::addr, guest_es_access_rights::name, guest_es_access_rights::exists());
+    dump_vmcs_field(guest_cs_access_rights::addr, guest_cs_access_rights::name, guest_cs_access_rights::exists());
+    dump_vmcs_field(guest_ss_access_rights::addr, guest_ss_access_rights::name, guest_ss_access_rights::exists());
+    dump_vmcs_field(guest_ds_access_rights::addr, guest_ds_access_rights::name, guest_ds_access_rights::exists());
+    dump_vmcs_field(guest_fs_access_rights::addr, guest_fs_access_rights::name, guest_fs_access_rights::exists());
+    dump_vmcs_field(guest_gs_access_rights::addr, guest_gs_access_rights::name, guest_gs_access_rights::exists());
+    dump_vmcs_field(guest_ldtr_access_rights::addr, guest_ldtr_access_rights::name, guest_ldtr_access_rights::exists());
+    dump_vmcs_field(guest_tr_access_rights::addr, guest_tr_access_rights::name, guest_tr_access_rights::exists());
+    dump_vmcs_field(guest_interruptibility_state::addr, guest_interruptibility_state::name, guest_interruptibility_state::exists());
+    dump_vmcs_field(guest_activity_state::addr, guest_activity_state::name, guest_activity_state::exists());
+    dump_vmcs_field(guest_smbase::addr, guest_smbase::name, guest_smbase::exists());
+    dump_vmcs_field(guest_ia32_sysenter_cs::addr, guest_ia32_sysenter_cs::name, guest_ia32_sysenter_cs::exists());
+    dump_vmcs_field(vmx_preemption_timer_value::addr, vmx_preemption_timer_value::name, vmx_preemption_timer_value::exists());
 }
 
 void
-vmcs_intel_x64::dump_vmcs_32bit_host_state()
+debug::dump_32bit_host_state_field()
 {
     bfdebug << bfendl;
     bfdebug << "32bit Host State Fields:" << bfendl;
-    PRINT_FIELD(true, VMCS_HOST_IA32_SYSENTER_CS);
+    dump_vmcs_field(host_ia32_sysenter_cs::addr, host_ia32_sysenter_cs::name, host_ia32_sysenter_cs::exists());
 }
 
 void
-vmcs_intel_x64::dump_vmcs_natural_control_state()
+debug::dump_natural_width_control_fields()
 {
     bfdebug << bfendl;
     bfdebug << "Natural Width Control Fields:" << bfendl;
-    PRINT_FIELD(true, VMCS_CR0_GUEST_HOST_MASK);
-    PRINT_FIELD(true, VMCS_CR4_GUEST_HOST_MASK);
-    PRINT_FIELD(true, VMCS_CR0_READ_SHADOW);
-    PRINT_FIELD(true, VMCS_CR4_READ_SHADOW);
-    PRINT_FIELD(true, VMCS_CR3_TARGET_VALUE_0);
-    PRINT_FIELD(true, VMCS_CR3_TARGET_VALUE_1);
-    PRINT_FIELD(true, VMCS_CR3_TARGET_VALUE_2);
-    PRINT_FIELD(true, VMCS_CR3_TARGET_VALUE_31);
+    dump_vmcs_field(cr0_guest_host_mask::addr, cr0_guest_host_mask::name, cr0_guest_host_mask::exists());
```

Comment:
I like how you added newlines to the other fields since they were really large. I would keep the same thing for all of the fields, so this would be

```cpp
dump_vmcs_field(cr0_guest_host_mask::addr,
                              cr0_guest_host_mask::name, 
                              cr0_guest_host_mask::exists());
```